lib/plugin/css-variables-collector.js

```diff
@@ -168,29 +199,13 @@ CSSVariablesCollectorPlugin.prototype = {
 		return node;
 	},
 
-	visitImport(node, visitArgs) {
-		// store the import context
-		this.importStack.push(node);
-		return node;
-	},
-
-	visitImportOut(node) {
-		// remove import context
-		this.importStack.pop();
-		return node;
-	},
-
-	visitRuleset(node, visitArgs) {
-		node.rules.forEach((value) => {
-			const isVarDeclaration = value instanceof less.tree.Rule && typeof value.name === "string" && value.name.startsWith("@");
-			if (!this._isInMixinOrParen() && isVarDeclaration) {
-				// add the variable declaration to the list of vars
-				this.vars[value.name.substr(1)] = {
-					css: this._getCSS(value.value),
-					export: !this._isInGlobalOrBaseImport()
-				};
-			}
-		});
+	visitUrl(node, visitArgs) {
+		// we mark the less variables which should be updated after eval
```

Comment:
My expectation was actually the other way round.
To use the "override" values (from the new preVisitor) in general, and only use the current ones for the variable references (which was the main reason for enabling the preEvalVisitor mode).
Otherwise I'd expect that also some other values might differ unexpectedly.

Reply:
There are issues within the isPreEval phase where less variables with the value none also have the URL content - so doing it this way is the best to determine properly the variables and apply only the needed later! All other options require much more code!

Reply:
Got it. Would be great to have those specialities documented somewhere.

Reply:
It is described in my comments